test/query_test.rb

```diff
@@ -32,9 +39,21 @@ def test_find
     assert_equal [], User.find(name: "Noah").to_a
   end
 
+  def test_find!
+    user = User.create(name: "Bryan")
+    assert_equal user, User.find!(user.id)
+    assert_raises(Minidoc::DocumentNotFoundError) { User.find!(BSON::ObjectId.new) }
```

Comment:
That assertion is true (it would raise, and therefore pass), AFAIK.

The fact that find takes id-or-selector is a poor design IMO, but I'm not correcting that here. The intent was that `x!` should be `x or raise`, so that's as far as I've tested.